code/__HELPERS/unsorted.dm

```diff
@@ -1423,6 +1423,11 @@ B --><-- A
 /atom/movable/proc/orbit(atom/A, radius = 10, clockwise = 1, angle_increment = 15)
 	if(!istype(A))
 		return
+	if (orbiting)
+		stop_orbit()
+		sleep(1) //sadly this is the only way to ensure the original orbit proc stops and resets the atom's transform.
+		if (orbiting || !istype(A)) //post sleep re-check
+			return 

```

Comment:
I'll admit that being qdeleted within a decisecond is unlikely, but it could happen